### PR TITLE
feat(MgvReader): Update market order simulation to match new tick-based algorithm

### PR DESCRIPTION
**Pull request by espendk**

### CODE REVIEW

**Review by adhusson — APPROVED** [review]
looks good to me.

I'm trying to think of a way to let Mangrove simulate that. The only part that needs changing is the "transfer tokens" part, right?

**Comment by espendk** [comment]
> I'm trying to think of a way to let Mangrove simulate that. The only part that needs changing is the "transfer tokens" part, right?

Umm, not sure. Any state change needs to be avoided, including not updating the offer list :-)

Maybe we could execute the order and then revert with the result?

But I'm not sure this should be a priority - I'm unsure if this feature will end up being used in practice.

File: src/periphery/MgvReader.sol [hunk 1]
```diff
@@ -256,50 +283,42 @@ contract MgvReader {
...
-    external
-    view
-    returns (VolumeData[] memory)
-  {
-    return marketOrder(olKey, takerWants, takerGives, fillWants, true);
-  }
-
-  function internalMarketOrder(MarketOrder memory mr, bool proceed) internal view {
+  function internalMarketOrder(MarketOrder memory mr) internal view {
+    console.log("internalMO");
```
Comment (lnist) [inline src/periphery/MgvReader.sol:298]: are these logs intentional?

**Review by lnist — COMMENTED** [review]
This saves a lot of gas compared to a call to the actual market order - and does not require tokens for transfers.  But it does come at the cost of logic duplication and imprecision due to not executing the real order.
I am not sure what scenario we see it used, but for now it is probably OK.

File: src/periphery/MgvReader.sol (hunk 1 above)
Comment (espendk) [reply src/periphery/MgvReader.sol:298]: doh! I'll remove them in another PR

Comment (espendk) [reply src/periphery/MgvReader.sol:298]: PR to fix this here: https://github.com/mangrovedao/mangrove-core/pull/507

**Pull request merged**